### ISSUE TITLE
Error in com.google.api.ads.adwords.lib.utils.ReportDownloadResponse class

### ISSUE THREAD

**Issue by AlexanderDahmen** [googleads/googleads-java-lib#150]
Hello,

I am using the googleads-java-lib for downloading AdWords Reports with a Java client.
For that, I am using ads-lib-3.16 with the following pom.xml:

```xml
        <dependency>
            <groupId>com.google.api-ads</groupId>
            <artifactId>ads-lib</artifactId>
            <version>3.16.0</version>
        </dependency>
        <dependency>
            <groupId>com.google.api-ads</groupId>
            <artifactId>adwords-axis</artifactId>
            <version>3.16.0</version>
        </dependency>
```

The error I encountered recently after updating the dependencies was this Exception:
```
Caused by: java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkArgument(ZLjava/lang/String;I)V
   at com.google.api.ads.adwords.lib.utils.ReportDownloadResponse.<init>(ReportDownloadResponse.java:47) ~[ads-lib-3.15.1.jar!/:na]
   at com.google.api.ads.adwords.lib.utils.AdHocReportDownloadHelperImpl.handleResponse(AdHocReportDownloadHelperImpl.java:104) ~[ads-lib-3.15.1.jar!/:na]
   at com.google.api.ads.adwords.lib.utils.AdHocReportDownloadHelperImpl.downloadReport(AdHocReportDownloadHelperImpl.java:60) ~[ads-lib-3.15.1.jar!/:na]
   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_171]
   at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_171]
   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_171]
   at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_171]
   at com.google.api.ads.common.lib.utils.AdsUtilityInvocationHandler.handleInvocation(AdsUtilityInvocationHandler.java:46) ~[ads-lib-3.15.1.jar!/:na]
   at com.google.common.reflect.AbstractInvocationHandler.invoke(AbstractInvocationHandler.java:87) ~[guava-18.0.jar!/:na]
   at com.sun.proxy.$Proxy155.downloadReport(Unknown Source) ~[na:na]
   at com.google.api.ads.adwords.lib.utils.AdHocReportDownloadHelper.downloadReport(AdHocReportDownloadHelper.java:46) ~[ads-lib-3.15.1.jar!/:na]
   at com.google.api.ads.adwords.lib.utils.v201802.ReportDownloader.downloadReport(ReportDownloader.java:65) ~[ads-lib-
   at <<<my calling code>>>>
   ...
```

When I checked the ads-lib-3.16.jar in my maven repository with downloaded sources, the indicated code looks like this:
```java
Preconditions.checkArgument(rawResponse.getHttpStatus() == HttpURLConnection.HTTP_OK,
        "Expected a successful raw response but raw response status is: %s",
        rawResponse.getHttpStatus());
```

But upon checking the **Bytecode** of the class, the call to `Preconditions.checkArgument` is this:
```
`* 38: invokestatic  com/google/common/base/Preconditions.checkArgument:(ZLjava/lang/String;I)V`
```

I also confirmed this opening the `ReportDownloadResponse.class` file from the jar, where it does indeed call `checkArgument[byte][byte][byte](ZLjava/lang/String;I)V`.

Could you possibly fix this errornous call?
Thank you and have a nice day,
 - Alexander Dahmen

**Comment by AlexanderDahmen** [comment]
I just noticed that I got the Stacktrace from version 3.15.1, but I just confirmed the error to still be in 3.16:
```java

    public ReportDownloadResponse(RawReportDownloadResponse rawResponse) {
        // <editor-fold defaultstate="collapsed" desc="Compiled Code">
        /* 0: aload_0
         * 1: invokespecial java/lang/Object."<init>":()V
         * .
         * .
         * .
         * >>> The error is here:
         * 38: invokestatic  com/google/common/base/Preconditions.checkArgument:(ZLjava/lang/String;I)V
         * 41: return
         *  */
        // </editor-fold>
    }
```

The guava-20.0.jar file bundled in the releases libs does not contain the method `Preconditions.checkArgument(String)`, all method signatures start with a boolean parameter:

```java
    public static void checkArgument(boolean expression) {
        // Compiled Code
    }

    public static void checkArgument(boolean bln, @javax.annotation.Nullable Object o) {
        // Compiled Code
    }

   // Other overloads ...
```

**Comment by nwbirnie** [comment]
Hey Alex,

Can you point me at the specific artifact which is calling checkArgument(String) please?

Also could you advise which version of the JDK you are running maven with please?

Best wishes,

Nick

**Comment by AlexanderDahmen** [comment]
Hello Nick,

Great Timing. I was looking into the dependencies and library jars, and found something just half an hour before your answer.
Turns out the reason it broke was not because of ads-lib or even guava, but because I had an embedded-redis library for integration testing, which forced the guava version down to `guava-18.0` - which caused the call to `Preconditions.checkArgument` to fail.

Furthermore, I interpreted the bytecode wrong: the **Z** in `(ZLjava/lang/String;I)` represents a boolean, so there is no problem in that call.

My bad, this is not an issue with your library, consider this problem solved.
Thank you very much for your time!
